checkout_and_build_pr.sh

```diff
@@ -0,0 +1,217 @@
+    if [ "$EXIT_CODE" = 0 ] ; then
+        REPO_CMD_HISTORY+="\033[1;32m     SUCCESS: running command [${CMD_TO_EXECUTE}], exit code ${EXIT_CODE} \n\033[0m"
+    else
+        if [ "$FAILURE_UNEXPECTED" = true ] ; then
+            REPO_CMD_HISTORY+="\033[1;31m     ERROR: running [${CMD_TO_EXECUTE}], exit code ${EXIT_CODE} \n\033[0m"
+        else
+            REPO_CMD_HISTORY+="\033[0;33m     INFO: running [${CMD_TO_EXECUTE}], exit code ${EXIT_CODE} \n\033[0m"
+        fi
+    fi
+}
+
+
+function checkout_repo_branch() {
+    REPO_NAME=$1
+    BRANCH_NAME_TO_CHECKOUT=$2
+    GIT_SSH="git@github.com:ONSdigital/${REPO_NAME}.git"
+
+    REPO_CMD_HISTORY+="REPO: ${REPO_NAME} \n"
+
+    # echo "Getting cloning ${GIT_SSH}"
+    execute_and_record_command "git clone ${GIT_SSH} " true
+    cd $REPO_NAME
+
+    execute_and_record_command "git checkout ${BRANCH_NAME_TO_CHECKOUT}" false
+}
+
+function checkout_and_build_repo_branch() {
+    REPO_NAME=$1
+    BRANCH_NAME_TO_CHECKOUT=$2
+    RUN_TESTS_CMD=$3
+    SKIP_TESTS_CMD=$4
+
+    checkout_repo_branch $REPO_NAME $BRANCH_NAME_TO_CHECKOUT
+
+    if [ "$SKIP_TESTS" = true ] ; then
+        echo "${REPO_NAME} Skipping Tests: Running CMD: ${SKIP_TESTS_CMD}"
+        execute_and_record_command "${SKIP_TESTS_CMD}" true        
+    else
+        echo "${REPO_NAME} Running Tests: Running CMD: ${RUN_TESTS_CMD}"
+        execute_and_record_command "${RUN_TESTS_CMD}" true  
+    fi
+}
+
+function killOffRunningDocker() {
+
+    # Not sure if you'd never want to do this?  Doesn't cost any time if already clear
+    if [ "$KILL_DOCKER" = false ] ; then
+        echo "Leaving Docker alone"
+    else
+        echo 'Stopping any running Docker containers'
+        docker stop $(docker ps -aq)
+
+        echo 'Removing any stopped Docker containers'
+        docker rm $(docker ps -a -q)
+    fi
+
+}
+
+function createNewBaseDir() {
+    BRANCH_DIR_TO_MAKE=$1
+    BASE_DIR=$2
+
+    echo "Passed branch dir to make ${BRANCH_DIR_TO_MAKE}"
+
+    BASE_DIR="${BASE_DIR}/${BRANCH_DIR_TO_MAKE}"
+
+    # TODO:  What to do if dir alerady exists? rm it, fail etc.. 
+
+    echo "Making new base dir ${BASE_DIR}"
+    mkdir $BASE_DIR
+
+    cd $BASE_DIR
+    echo "Now in new DIR: ${PWD}"
+}
+
+
+########################################################################################################################
+#
+#         START OF SCRIPT
+#
+########################################################################################################################
+
+# Internal Variable, will use to record time
+SECONDS=0
+
+# Check Branch name is set
+if [ -z "$BRANCH_NAME" ]; then
+    echo "You Must set BRANCH_NAME=<branch to test>, use main, if you want main"
+    exit 2;
+fi
+
+# Base Dir to Checkout To
+if [ -z "$BASE_DIR" ]; then
+    echo "You Must set BASE_DIR=<your base dir>.  BASE_DIR does not need the branch name, that will be created"
+    exit 2;
+fi
```

Comment:
Not mentioned in the documentation?

Reply:
It does build to the branch name, but under the supplied BASE_DIR.
I originally had it effectively ../../<branch_name>.   Don't fancy it building them all under my main docker-dev
Suggest we discuss and enhance this as required